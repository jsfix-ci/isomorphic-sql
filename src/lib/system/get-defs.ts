import { columnColumnNames } from "../constant/column-column-names";
import { tableColumnNames } from '../constant/table-column-names';
import { newline } from "../constant/delims";
const column = 'c';
const table = 't';

export const getDefs = (schemaname = 'public') => ({
    sql: [
        `   SELECT ${columnColumnNames.map(c => `${column}.${c}`)}, ${tableColumnNames.map(t => `${table}.${t}`)}`,
        `   FROM pg_catalog.pg_tables ${table}`,
        `   INNER JOIN information_schema.columns ${column} on ${column}.table_name = ${table}.tablename`,
        `   WHERE schemaname = '${schemaname}'`,
        `   ORDER BY tablename`,
    ].join(newline),
    columns: [...columnColumnNames, ...tableColumnNames]
});