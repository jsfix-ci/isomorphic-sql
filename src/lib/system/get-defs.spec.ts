import { getDefs } from './get-defs';

describe('getDefs', () => {
    it('returns a sql string', () => {
        expect(getDefs()).toMatchSnapshot();
    });
});