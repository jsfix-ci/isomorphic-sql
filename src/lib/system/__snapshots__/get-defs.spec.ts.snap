// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDefs returns a sql string 1`] = `
Object {
  "columns": Array [
    "column_name",
    "data_type",
    "is_nullable",
    "column_default",
    "schemaname",
    "tablename",
    "tableowner",
    "tablespace",
    "hasindexes",
    "hasrules",
    "hastriggers",
    "rowsecurity",
  ],
  "sql": "   SELECT c.column_name,c.data_type,c.is_nullable,c.column_default, t.schemaname,t.tablename,t.tableowner,t.tablespace,t.hasindexes,t.hasrules,t.hastriggers,t.rowsecurity
    FROM pg_catalog.pg_tables t
    INNER JOIN information_schema.columns c on c.table_name = t.tablename
    WHERE schemaname = 'public'
    ORDER BY tablename",
}
`;
